spec.html

```diff
@@ -36892,7 +36892,8 @@ <h1>RegExp.prototype [ @@match ] ( _string_ )</h1>
             1. Return ? RegExpExec(_rx_, _S_).
           1. Else,
             1. Assert: _global_ is *true*.
-            1. Let _fullUnicode_ be ! ToBoolean(? Get(_rx_, *"unicode"*)).
+            1. Let _fullUnicode_ be ! ToBoolean(? Get(_rx_, *"unicodeSets"*)).
+            1. If _fullUnicode_ is *false*, set _fullUnicode_ to ! ToBoolean(? Get(_rx_, *"unicode"*)).
```

Comment:
the conditional Get seems odd; why not get both unconditionally?

Reply:
https://github.com/mathiasbynens/ecma262/pull/13#discussion_r874721400

Reply:
TL;DR Why waste CPU cycles accessing a getter unnecessarily?

Reply:
Because it’s deterministic, and for true regexes wouldn’t be observable or slow anyways.

Although it’s incredibly unlikely, i can also see a regex subclass relying on the getter being called and suddenly breaking when someone instantiates it with the v flag.

Reply:
> Although it’s incredibly unlikely, i can also see a regex subclass relying on the getter being called and suddenly breaking when someone instantiates it with the v flag.

Why would a subclass rely on the `unicode` getter being called for non-`u` instances? That code would already be broken today, with or without the `v` flag.

Reply:
Yeah the contrived example i was reading for isn’t very compelling regardless. The main thing is that i prefer deterministic Gets that aren’t dependent on the value.

Reply:
I didn't say it was an invariant, I said it was a preference. I recognize there's existing branching; i prefer to avoid additional branching.

Reply:
I’m struggling to understand this preference.

**Option A (current patch) — only calling getters when necessary:**

- consistent with current practice of branching carefully
- efficient

**Option B (Jordan’s preference) — unconditionally calling both getters for global regular expressions:**

- deviates from the precedent of branching where applicable / only calling what’s necessary
- less efficient

Option A seems strictly better than option B.

Reply:
Other parts of this function actually use
```
If _flags_ contains *"u"* or _flags_ contains *"v"*, let _fullUnicode_ be *true*; else let _fullUnicode_ be *false*.
```
rather than Get()ing each flag each time. Shouldn't this part of the same function use that same construction?

Reply:
*Update:* We’re awaiting the outcome of https://github.com/tc39/ecma262/pull/2791 before taking further steps here.

Reply:
Looks like #2791 has consensus, so we can go ahead and update this PR accordingly (without necessarily waiting for it to land). @markusicu Is this something you wanna take on?

Reply:
> Looks like #2791 has consensus, so we can go ahead and update this PR accordingly (without necessarily waiting for it to land). @markusicu Is this something you wanna take on?

I made a note but I am not sure when I will get to it.

Reply:
https://github.com/tc39/ecma262/pull/2791 has landed since.

Reply:
PR: https://github.com/mathiasbynens/ecma262/pull/16